corehq/apps/users/role_utils.py

```diff
@@ -126,3 +127,21 @@ def enable_attendance_coordinator_role_for_domain(domain):
     if role.is_archived:
         role.is_archived = False
         role.save()
+
+
+def get_commcare_analytics_access_for_user_domain(couch_user, domain):
+    domain_membership = couch_user.get_domain_membership(domain)
+    is_admin = domain_membership.is_admin
+
+    domain_role_permissions = domain_membership.role.permissions
```

Comment:
nit: It seems you can avoid a db query for role by returning from here itself for admins without loading the permissions.

Reply:
@mkangia You mean something like

>
    if is_admin:
        permissions = {
            "can_edit": True,
            "can_view": True,
        }
        cca_roles = COMMCARE_ANALYTICS_USER_ROLES
    else:
        domain_role_permissions = domain_membership.role.permissions
        permissions = {
            "can_edit": is_admin or domain_role_permissions.edit_commcare_analytics,
            "can_view": is_admin or domain_role_permissions.view_commcare_analytics,
        }
        cca_roles = domain_role_permissions.commcare_analytics_roles

Reply:
Yes :+1: